### PR TITLE
Release 2.5.2

### PR DESCRIPTION
**Pull request by MaxUlysse**
Many thanks to contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** [guidelines](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -40,36 +40,36 @@ jobs:
     environment:
       GENOME: GRCh37
       SPECIES: homo_sapiens
-      VEP_VERSION: "95"
+      VEP_VERSION: "98"
```
Comment (MaxUlysse) [inline .circleci/config.yml]:
```suggestion
      VEP_VERSION: "95"
```

Comment (MaxUlysse) [inline .circleci/config.yml]:
```suggestion
      VEP_VERSION: "95"
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -40,36 +40,36 @@ jobs:
...
-          command: echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin ; docker push nfcore/sarekvep:2.5.${GENOME}
+          command: echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin ; docker push nfcore/sarekvep:2.5.2.${GENOME}
 
   vepgrch38:
     << : *buildvep
     environment:
       GENOME: GRCh38
       SPECIES: homo_sapiens
-      VEP_VERSION: "95"
+      VEP_VERSION: "98"
```
Comment (MaxUlysse) [inline .circleci/config.yml]:
```suggestion
      VEP_VERSION: "95"
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -40,36 +40,36 @@ jobs:
...
 
   vepgrcm38:
     << : *buildvep
     environment:
       GENOME: GRCm38
       SPECIES: mus_musculus
-      VEP_VERSION: "95"
+      VEP_VERSION: "98"
```
Comment (MaxUlysse) [inline .circleci/config.yml]:
```suggestion
      VEP_VERSION: "95"
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -40,36 +40,36 @@ jobs:
...
 
   vepcanfam3_1:
     << : *buildvep
     environment:
       GENOME: CanFam3.1
       SPECIES: canis_familiaris
-      VEP_VERSION: "95"
+      VEP_VERSION: "98"
```
Comment (MaxUlysse) [inline .circleci/config.yml]:
```suggestion
      VEP_VERSION: "95"
```

File: docs/containers.md [hunk 5]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
 - Contain **[AlleleCount](https://github.com/cancerit/alleleCount)** 4.0.2
 - Contain **[BCFTools](https://github.com/samtools/bcftools)** 1.9
 - Contain **[BWA](https://github.com/lh3/bwa)** 0.7.17
-- Contain **[Control-FREEC](https://github.com/BoevaLab/FREEC)** 11.4
+- Contain **[Control-FREEC](https://github.com/BoevaLab/FREEC)** 11.5
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[Control-FREEC](https://github.com/BoevaLab/FREEC)** 11.4
```

File: docs/containers.md [hunk 6]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
 - Contain **[FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)** 0.11.8
-- Contain **[FreeBayes](https://github.com/ekg/freebayes)** 1.2.0
-- Contain **[GATK4](https://github.com/broadinstitute/gatk)** 4.1.2.0
+- Contain **[FreeBayes](https://github.com/ekg/freebayes)** 1.3.1
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[FreeBayes](https://github.com/ekg/freebayes)** 1.2.0
```

File: docs/containers.md [hunk 7]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
+- Contain **[GATK4](https://github.com/broadinstitute/gatk)** 4.1.4.0
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[GATK4](https://github.com/broadinstitute/gatk)** 4.1.2.0
```

File: docs/containers.md [hunk 8]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
 - Contain **[GeneSplicer](https://ccb.jhu.edu/software/genesplicer/)** 1.0
 - Contain **[HTSlib](https://github.com/samtools/htslib)** 1.9
-- Contain **[Manta](https://github.com/Illumina/manta)** 1.5.0
+- Contain **[Manta](https://github.com/Illumina/manta)** 1.6.0
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[Manta](https://github.com/Illumina/manta)** 1.5.0
```

File: docs/containers.md [hunk 9]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
 - Contain **[MultiQC](https://github.com/ewels/MultiQC/)** 1.7
-- Contain **[Qualimap](http://qualimap.bioinfo.cipf.es)** 2.2.2b
+- Contain **[Qualimap](http://qualimap.bioinfo.cipf.es)** 2.2.2c
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[Qualimap](http://qualimap.bioinfo.cipf.es)** 2.2.2b
```

File: docs/containers.md [hunk 10]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
 - Contain **[samtools](https://github.com/samtools/samtools)** 1.9
 - Contain **[snpEff](http://snpeff.sourceforge.net/)** 4.3.1t
 - Contain **[Strelka2](https://github.com/Illumina/strelka)** 2.9.10
-- Contain **[TIDDIT](https://github.com/SciLifeLab/TIDDIT)** 2.7.1
-- Contain **[VCFanno](https://github.com/brentp/vcfanno)** 0.3.1
+- Contain **[TIDDIT](https://github.com/SciLifeLab/TIDDIT)** 2.8.1
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[TIDDIT](https://github.com/SciLifeLab/TIDDIT)** 2.7.1
```

File: docs/containers.md [hunk 11]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
+- Contain **[VCFanno](https://github.com/brentp/vcfanno)** 0.3.2
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[VCFanno](https://github.com/brentp/vcfanno)** 0.3.1
```

File: docs/containers.md [hunk 12]
```diff
@@ -18,22 +18,22 @@ For annotation, the main container can be used, but the cache has to be download
...
 - Contain **[VCFtools](https://vcftools.github.io/index.html)** 0.1.16
-- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 95.2
+- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 98.2
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 95.2
```

File: docs/containers.md [hunk 13]
```diff
@@ -45,45 +45,99 @@ For annotation, the main container can be used, but the cache has to be download
 
 - Based on `nfcore/base:latest`
 - Contain **[GeneSplicer](https://ccb.jhu.edu/software/genesplicer/)** 1.0
-- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 95.2
+- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 98.2
```
Comment (MaxUlysse) [inline docs/containers.md]:
```suggestion
- Contain **[VEP](https://github.com/Ensembl/ensembl-vep)** 95.2
```

File: docs/output.md [hunk 14]
```diff
@@ -77,30 +102,42 @@ For further reading and documentation see the [data pre-processing workflow from
 For all samples:
 **Output directory: `results/Preprocessing/[SAMPLE]/Recalibrated`**
 
-* `[SAMPLE].recal.bam` and `[SAMPLE].recal.bai`
-  * BAM file and index
+- `[SAMPLE].recal.bam` and `[SAMPLE].recal.bai`
```
Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
- `[SAMPLE].recal.bam` and `[SAMPLE].recal.bam.bai`
```

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -18,7 +18,113 @@ jobs:
...
+        run: |
+          wget -qO- get.nextflow.io | bash
+          sudo mv nextflow /usr/local/bin/
+        env:
+          NXF_VER: '19.10.0'
+      - name: Download and tag image
+        run: |
+          docker pull nfcore/sarek:dev
+          docker tag nfcore/sarek:dev nfcore/sarek:2.5.2
+      - name: Run vriant calling test on specific tools
```
Comment (MaxUlysse) [inline .github/workflows/ci.yml]:
```suggestion
      - name: Run variant calling test on specific tools
```

File: .github/workflows/ci.yml [hunk 16]
```diff
@@ -18,7 +18,113 @@ jobs:
...
+          docker pull nfcore/sarek:dev
+          docker tag nfcore/sarek:dev nfcore/sarek:2.5.2
+      - name: Run targeted and splitfastq tests
+        run: |
+          nextflow run . -profile ${{ matrix.profile }},docker --verbose
+  tools:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        tool: [Haplotypecaller, Freebayes, Manta, mpileup, Mutect2, Strelka]
```
Comment (MaxUlysse) [inline .github/workflows/ci.yml]:
```suggestion
        tool: [Haplotypecaller, Freebayes, Manta, mpileup, Mutect2, Strelka, TIDDIT]
```

File: containers/vep/environment.yml [hunk 17]
```diff
@@ -1,11 +1,11 @@
...
-name: sarek-vep-2.5
+name: sarek-vep-2.5.2
 channels:
   - conda-forge
   - bioconda
   - defaults
 
 dependencies:
-  - ensembl-vep=95.2
+  - ensembl-vep=98.2
```
Comment (MaxUlysse) [inline containers/vep/environment.yml]:
```suggestion
  - ensembl-vep=95.2
```

File: containers/vep/Dockerfile [hunk 18]
```diff
@@ -7,12 +7,12 @@ LABEL \
...
 COPY environment.yml /
 RUN conda env create -f /environment.yml && conda clean -a
-ENV PATH /opt/conda/envs/sarek-vep-2.5/bin:$PATH
+ENV PATH /opt/conda/envs/sarek-vep-2.5.2/bin:$PATH
 
 # Setup default ARG variables
 ARG GENOME=GRCh38
 ARG SPECIES=homo_sapiens
-ARG VEP_VERSION=95
+ARG VEP_VERSION=98
```
Comment (MaxUlysse) [inline containers/vep/Dockerfile]:
```suggestion
ARG VEP_VERSION=95
```

**Comment by MaxUlysse** [comment]
Default being `GRCh38`, we're actually already using homo_sapiens for VEP:
https://github.com/nf-core/sarek/pull/78/files#diff-28b7bea57b04f15beeb4c1de0ee78382R251

**Comment by apeltzer** [comment]
Whats wrong with all these merge conflicts?

**Comment by MaxUlysse** [comment]
OK, so if you have complex merge conflicts and need to use command line to resolve them.
If you ever got this error:
```Bash
remote: Invalid username or password.
```
Just disable your 2FA and it'll work out fine

**Pull request merged**